File: jediterm-core-mpp/src/jvmMain/java/com/jediterm/terminal/model/StyleState.java
package com.jediterm.terminal.model;

import com.jediterm.terminal.TerminalColor;
import com.jediterm.terminal.TextStyle;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class StyleState {
  private volatile TextStyle myCurrentStyle = TextStyle.EMPTY;
  private volatile TextStyle myDefaultStyle = TextStyle.EMPTY;

  public StyleState() {
  }

  public @NotNull TextStyle getCurrent() {
    return myCurrentStyle;
  }

  public void reset() {
    myCurrentStyle = myDefaultStyle;
  }

  public void setDefaultStyle(@NotNull TextStyle defaultStyle) {
    myDefaultStyle = defaultStyle;
  }

  public @NotNull TerminalColor getDefaultBackground() {
    return Objects.requireNonNull(myDefaultStyle.getBackground());
  }

  public @NotNull TerminalColor getDefaultForeground() {
    return Objects.requireNonNull(myDefaultStyle.getForeground());
  }

  public void setCurrent(@NotNull TextStyle current) {
    myCurrentStyle = current;
  }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/terminal/model/StyleState.kt
package com.jediterm.terminal.model

import com.jediterm.terminal.TextStyle
import java.util.*
import kotlin.concurrent.Volatile

class StyleState {
    @Volatile
    var current: TextStyle = TextStyle.Companion.EMPTY

    @Volatile
    private var myDefaultStyle: TextStyle = TextStyle.Companion.EMPTY

    fun reset() {
        this.current = myDefaultStyle
    }

    fun setDefaultStyle(defaultStyle: TextStyle) {
        myDefaultStyle = defaultStyle
    }

    val defaultBackground: com.jediterm.terminal.TerminalColor
        get() = Objects.requireNonNull<com.jediterm.terminal.TerminalColor?>(myDefaultStyle.background)

    val defaultForeground: com.jediterm.terminal.TerminalColor
        get() = Objects.requireNonNull<com.jediterm.terminal.TerminalColor?>(myDefaultStyle.foreground)
}
